jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt

```diff
@@ -36,6 +37,25 @@ object SamTemplateUtils {
     private val MAPPER = ObjectMapper(YAMLFactory())
     private const val S3_URI_PREFIX = "s3://"
 
+    fun getFunctionEnvironmentVariables(template: Path, logicalId: String): Map<String, String> = readTemplate(template) {
+        val function = requiredAt("/Resources").get(logicalId)
+            ?: throw IllegalArgumentException("No resource with the logical ID $logicalId")
+        val globals = at("/Globals/Function/Environment/Variables")
+        val variables = function.at("/Properties/Environment/Variables")
+        val globalVars = try {
+            MAPPER.convertValue<Map<String, String>>(globals)
+        } catch (e: Exception) {
+            null
+        } ?: emptyMap()
+        val vars = try {
+            MAPPER.convertValue<Map<String, String>>(variables)
+        } catch (e: Exception) {
+            null
+        } ?: emptyMap()
```

Comment:
Can be simplified to `runCatching { MAPPER.convertValue<Map<String, String>>(variables) }.getOrDefault(emptyMap())`